sdk/lib/atl/atlimage.h

```diff
@@ -915,11 +915,13 @@ class CImage
         }
 
         // get procedure address of the DLL
-        template <typename TYPE>
-        TYPE AddrOf(const char *name)
+        template <typename FUN_T>
+        bool _getFUN(FUN_T& fun, const char *name)
         {
-            FARPROC proc = ::GetProcAddress(hinstGdiPlus, name);
-            return reinterpret_cast<TYPE>(proc);
+            if (fun)
+                return true;
+            fun = reinterpret_cast<FUN_T>(::GetProcAddress(hinstGdiPlus, name));
+            return fun != NULL;
```

Comment:
Not sure it's useful to return the bool since no checks are done where `_getFUN` is called below. It could just return the value of "fun" (and avoid the per-reference 1st parameter).

Reply:
The 1st parameter is provided to omit the template parameter.

Reply:
The return value is fixed.